api/v1alpha1/trafficmanagerprofile_types.go

```diff
@@ -0,0 +1,143 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type TrafficManagerProfile struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec TrafficManagerProfileSpec `json:"spec"`
+	// +optional
+	Status TrafficManagerProfileStatus `json:"status,omitempty"`
+}
+
+type TrafficManagerProfileSpec struct {
+	// Type of routing method.
+	// +kubebuilder:validation:Enum=Weighted
+	// +kubebuilder:default=Weighted
+	// +optional
+	RoutingMethod TrafficRoutingMethod
+
+	// The DNS settings of the Traffic Manager profile.
+	// +required
+	DNSConfig *DNSConfig
+
+	// The endpoint monitoring settings of the Traffic Manager profile.
+	MonitorConfig *MonitorConfig
+}
+
+type DNSConfig struct {
+	// +required
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=2147483647
+	// Traffic Manager allows you to configure the TTL used in Traffic Manager DNS responses to be as low as 0 seconds
+	// and as high as 2,147,483,647 seconds (the maximum range compliant with RFC-1035), enabling you to choose the value
+	// that best balances the needs of your application.
+	TTL *int64
+}
+
+type MonitorConfig struct {
+	// The monitor interval for endpoints in this profile. This is the interval at which Traffic Manager will check the health
```

Comment:
Also, I wonder if renaming it as "HealthProbeConfig" is better? k8s service also use this term.

Reply:
it's very traffic manager specific configuration. prefer to keep the original name.